packages/grafana-ui/src/components/Combobox/Combobox.tsx

```diff
@@ -146,11 +148,55 @@ export const Combobox = <T extends string | number>(props: ComboboxProps<T>) =>
         }
       }
 
-      baseSetItems(itemsToSet);
+      const reorderItemsByGroup = (items: Array<ComboboxOption<T>>) => {
+        const reorgItems: Array<ComboboxOption<T> | ComboboxOption<T>['group']> = [];
+        const processedGroups = new Set<ComboboxOption<T>['group']>();
+        const processedItems = new Set<ComboboxOption<T>>();
+        for (const item of items) {
+          if (processedItems.has(item)) {
+            continue;
+          }
+          // Add it to the list if the options does not have a group
+          if (typeof item === 'object' && !item.group) {
+            reorgItems.push(item);
+            processedItems.add(item);
+          } else {
+            const group = item.group;
+            // Add the group only if it is not already added
+            if (!processedGroups.has(group)) {
+              reorgItems.push(group);
+              processedGroups.add(group);
+            }
+            // Add the items of the group
+            const sameGroupItems = items.filter((i) => i.group === group && !processedItems.has(i));
+            reorgItems.push(...sameGroupItems);
+            sameGroupItems.forEach((i) => processedItems.add(i));
+          }
+        }
+        return reorgItems;
+      };
+
+      // Set the items including custom ones
+      baseSetItems(reorderItemsByGroup(itemsToSet));
     },
     [createCustomValue, id, ariaLabelledBy]
   );
 
+  const isLastItemGroup = (
```

Comment:
This is mainly used for styles, right? Can you use `:last-child` CSS selector instead?

Reply:
Not really, as groups are rows as the options; they are not nested.